account_move_reconcile_helper/models/account_move_line.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMoveLine(models.Model):
+
+    _inherit = 'account.move.line'
+
+    partial_reconciliation_in_progress = fields.Boolean(
+        compute='_compute_partial_reconciliation_in_progress')
+    reconcile_line_ids = fields.One2many(
+        compute='_compute_reconciled_lines',
+        comodel_name='account.move.line',
+        string="Reconciled lines")
+
+    @api.multi
+    @api.depends('matched_debit_ids', 'matched_credit_ids')
+    def _compute_partial_reconciliation_in_progress(self):
+        for rec in self:
+            rec.partial_reconciliation_in_progress = (
+                bool(rec.matched_debit_ids) or bool(rec.matched_credit_ids))
+
+    @api.multi
+    def _compute_reconciled_lines(self):
+        for rec in self:
+            rec.reconcile_line_ids = rec._get_reconciled_lines()
+
+    @api.multi
+    def _get_reconciled_lines(self, move_lines=None):
+        """
+        Returns lines which were reconciled directly or indirectly with
+        current lines given in self.
+
+        If A has been reconciled (or partially) with B, and B with C. This
+        method will returns A, B, and C.
+
+        :param move_lines: found moves lines to avoid recursivity
+        :return: recordset('account.move.line')
+        """
+        move_lines = move_lines or self.env[self._name]
+
+        for line in self:
+            if line.full_reconcile_id:
+                partial_reconciles = line.full_reconcile_id.mapped(
+                    'partial_reconcile_ids')
+                matched_lines = (
+                    partial_reconciles.mapped('debit_move_id') |
+                    partial_reconciles.mapped('credit_move_id'))
+            elif line.credit > 0:
+                matched_lines = line.matched_debit_ids.mapped('debit_move_id')
+            else:
+                matched_lines = line.matched_credit_ids.mapped(
+                    'credit_move_id')
+
+            if not matched_lines:
+                continue
+
+            move_lines |= line
+
+            for matched_line in matched_lines:
+                if matched_line not in move_lines:
+                    move_lines |= matched_line
+                    move_lines |= matched_line._get_reconciled_lines(
+                        move_lines)
+
+        return move_lines
```

Comment:
I'm wondering if this could be simplified using `full_reconcile_id.reconciled_line_ids` and an inner function for the recursion?

```python
def _get_reconciled_lines(self):
        move_lines = self.env[self._name]

	def _add(lines):
	    for line in lines:
		if line in move_lines:
                    continue
	        move_lines |= line
	        _add(move_lines.matched_debit_ids)
	        _add(move_lines.matched_credit_ids)

	if line.full_reconcile_id:
	    move_lines |= line.full_reconcile_id.reconciled_line_ids
	else:
	   _add(self)

        return move_lines
```